.github/workflows/test-template.yml

```diff
@@ -130,13 +130,12 @@ jobs:
       - name: Code checkout
         uses: actions/checkout@v4
 
-      - name: Install NodeJS ${{ inputs.node-version }}
-        uses: actions/setup-node@v4
+      - name: Install Node ${{ inputs.node-version }} and Dependencies
+        uses: ./.github/actions/npm-setup
         with:
+          runner: ${{ inputs.runner }}
           node-version: ${{ inputs.node-version }}
-
-      - name: Install dependencies
-        run: npm ci --omit=optional
+          workspace: "${{ inputs.workspace }}"
```

Comment:
@cristianrgreco I have one question here, probably caused by my lack of knowledge in the build system used here.

IIUC each module will be built in a worker, but should they reuse any cached node_modules dir from a previous stage? I'm adding a cache key that uses the workspace dir, and it could not be that performant if each module needs to build their own node_modules for first time. Of course, if the package-json.lock file does not change, it will be reused across builds, but in any case, I'd like to to tune this in the right manner.

Reply:
See this error in chromadb: https://github.com/testcontainers/testcontainers-node/actions/runs/13686730534/job/38272880849

Reply:
@mdelapenya The project is structured using [NPM workspaces](https://docs.npmjs.com/cli/v8/using-npm/workspaces). The `testcontainers` package is built standalone, and each module has a dependency on the `testcontainers` package **plus** its own dependencies. So chromaDB for example depends on `testcontainers` + `chromadb`, for example. I'm not aware of a way to merge dependencies between packages. I think the best we'll get is a cache for `testcontainers` and a cache for each module.

IDK if it's worth also looking at switching from NPM to PNPM, I hear it's faster at resolving/installing packages.

Reply:
I think I'm adding a cache for testcontainers and a cache per module 🤔 If you check out https://github.com/testcontainers/testcontainers-node/actions/runs/13686730534/job/38272872166, for chroma with NodeJS 18.x, the cache key is `refactor-pipeline-ubuntu-22.04-node-18.x-packages/modules/chromadb-cf7a257f906333ec16e5ea58a4b07b39c48331b08ee6d1250a6b4f61f56ffff9`, which is the same used by the failing pipeline: `refactor-pipeline-ubuntu-22.04-node-18.x-packages/modules/chromadb-cf7a257f906333ec16e5ea58a4b07b39c48331b08ee6d1250a6b4f61f56ffff9`.

I wonder if we should cache more paths, and not just node_modules. Maybe the `npm cache dir` too?

Reply:
Yep the cache keys are correct. 

> I wonder if we should cache more paths, and not just node_modules

Yes I think that's the problem. There's `node_modules` in:
- `/node_modules`
- `/packages/testcontainers/node_modules`
- `packages/modules/*/node_modules`